cudamapper/tests/Test_CudamapperIndexGPUTwoIndices.cu

```diff
@@ -0,0 +1,1333 @@
+    expected_new_value_mask_h.push_back(0);
+    representations_h.push_back(8);
+    expected_new_value_mask_h.push_back(1);
+    representations_h.push_back(8);
+    expected_new_value_mask_h.push_back(0);
+    representations_h.push_back(8);
+    expected_new_value_mask_h.push_back(0);
+    representations_h.push_back(9);
+    expected_new_value_mask_h.push_back(1);
+    representations_h.push_back(9);
+    expected_new_value_mask_h.push_back(0);
+    representations_h.push_back(9);
+    expected_new_value_mask_h.push_back(0);
+
+    std::uint32_t number_of_threads = 3;
+
+    test_create_new_value_mask(representations_h,
+                               expected_new_value_mask_h,
+                               number_of_threads);
+}
+
+TEST(TestCudamapperIndexGPUTwoIndices, test_create_new_value_mask_small_data_large_example)
+{
+    const std::uint64_t total_sketch_elements                    = 10000000;
+    const std::uint32_t sketch_elements_with_same_representation = 1000;
+
+    thrust::host_vector<representation_t> representations_h;
+    thrust::host_vector<std::uint32_t> expected_new_value_mask_h;
+    for (std::size_t i = 0; i < total_sketch_elements; ++i)
+    {
+        representations_h.push_back(i / sketch_elements_with_same_representation);
+        if (i % sketch_elements_with_same_representation == 0)
+            expected_new_value_mask_h.push_back(1);
+        else
+            expected_new_value_mask_h.push_back(0);
+    }
+
+    std::uint32_t number_of_threads = 256;
+
+    test_create_new_value_mask(representations_h,
+                               expected_new_value_mask_h,
+                               number_of_threads);
+}
+
+// ************ Test find_first_occurrences_of_representations_kernel **************
+
+void test_find_first_occurrences_of_representations_kernel(const thrust::host_vector<std::uint64_t>& representation_index_mask_h,
+                                                           const thrust::host_vector<representation_t>& input_representations_h,
+                                                           const thrust::host_vector<std::uint32_t>& expected_starting_index_of_each_representation_h,
+                                                           const thrust::host_vector<representation_t>& expected_unique_representations_h,
+                                                           const std::uint32_t number_of_threads)
+{
+    const thrust::device_vector<std::uint64_t> representation_index_mask_d(representation_index_mask_h);
+    const thrust::device_vector<representation_t> input_representations_d(input_representations_h);
+    ASSERT_EQ(expected_starting_index_of_each_representation_h.size(), representation_index_mask_h.back());
+    ASSERT_EQ(expected_unique_representations_h.size(), representation_index_mask_h.back());
+    thrust::device_vector<std::uint32_t> starting_index_of_each_representation_d(expected_starting_index_of_each_representation_h.size());
```

Comment:
Optional: I would set the sizes based on the input data, not on the expected output. That explains the relation of the different quantities a bit better than setting the size to the expected length.
Also, otherwise the `ASSERT_EQ(starting_ind..._d(expected_starting_ind..._h.size())` further down doesn't make much sense.

Reply:
True. Using `const std::uint64_t number_of_unique_representations = representation_index_mask_h.back();` instead.

I left the `ASSERT_EQ(starting_ind..._d(expected_starting_ind..._h.size()) ` checks below as they make sure that nothing unexpected happened in test body (for example someone altering some code by mistake) and as those checks are very cheap they should not have any influence on the test runtime